### PR TITLE
docs: adding GMP examples code

### PR DESCRIPTION
**Pull request by gsk967**
## Description

In this PR we are adding example code for deploying smart contract evm chain and transfer some `aUSDC` from evm chain to umee chain using axelar GMP

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

## Summary by CodeRabbit

- **New Features**
	- Enabled cross-chain transfers between Fantom EVM and Umee chains using Axelar General Message Passing (GMP) for enhanced interoperability.

### CODE REVIEW

File: x/uibc/gmp/README.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
+# GMP Handler
+
+Cross chain transfers using Axelar General Message Passing (GMP)
+More info about cosmos gmp find [here](https://docs.axelar.dev/dev/cosmos-gmp)
+
+## Example transfer from fantom evm to umee using Axelar GMP
```
Comment (robert-zaremba) [inline x/uibc/gmp/README.md:6]:
```suggestion
## Example transfer from fantom EVM to umee using Axelar GMP
```

File: x/uibc/gmp/README.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
+# GMP Handler
+
+Cross chain transfers using Axelar General Message Passing (GMP)
+More info about cosmos gmp find [here](https://docs.axelar.dev/dev/cosmos-gmp)
```
Comment (robert-zaremba) [inline x/uibc/gmp/README.md:4]:
```suggestion
More info about cosmos GMP find [here](https://docs.axelar.dev/dev/cosmos-gmp)
```

File: x/uibc/gmp/README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+## Example transfer from fantom evm to umee using Axelar GMP
+
+```bash
+# clone the repo
+$ git clone https://github.com/axelarnetwork/evm-cosmos-gmp-sample.git
+$ cd evm-cosmos-gmp-sample/native-integration/multi-send/solidity
+## Build the contracts on this repo
+$ npm run build
```
Comment (robert-zaremba) [inline x/uibc/gmp/README.md:13]:
```suggestion
```bash
# clone the repo
git clone https://github.com/axelarnetwork/evm-cosmos-gmp-sample.git
cd evm-cosmos-gmp-sample/native-integration/multi-send/solidity

## Build the contracts on this repo
npm run build
```

**Review by robert-zaremba — COMMENTED** [review]
Please provide the details, with link to the expected proto structure (proto message), what to put in the message. Also explain that GMP messages are handled by ICS-20 memo and link the code (file, without the line number - code lines can change)

**Comment by gsk967** [comment]
> Please provide the details, with link to the expected proto structure (proto message), what to put in the message. Also explain that GMP messages are handled by ICS-20 memo and link the code (file, without the line number - code lines can change)

We did not tested phase-2 memo handler , once we tested with smart contract  I will update the docs

File: x/uibc/gmp/README.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    const factory = ContractFactory.fromSolidity(MultiSend, wallet);
+
+    const contract = await factory.deploy(fantom.gateway, fantom.gasService)
+    const tx = await contract.deployed();
+
+    console.log(`multi send contract deployed on ${tx.address}`);
+})();
+```
+
+### Deploy the contract
```
Comment (robert-zaremba) [inline x/uibc/gmp/README.md:121]:
```suggestion
Deploy the contract
```

File: x/uibc/gmp/README.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    SourceChain   string `json:"source_chain"`
+    SourceAddress string `json:"source_address"`
+    Payload       []byte `json:"payload"`
+    Type          int64  `json:"type"`
+}
+```
+
+When user wants to execute leverage supply, supply collateral or liquidate transaction using Axelar GMP , user have to send `Payload=[]byte(Msg)` in `Memo` ,
+in below example contract change [payload](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/blob/main/native-integration/multi-send/solidity/contracts/MultiSend.sol) feild and version ,for now we are supporting `uint32(2)` version.
+
```
[the remark this answers is not in the document]
Comment (gsk967) [reply x/uibc/gmp/README.md:19]: I will do this on separate task

[the remark this answers is not in the document]
Comment (gsk967) [reply x/uibc/gmp/README.md:12]: https://linear.app/umee/issue/UMEE-1129/ibc-hooks-evm-example , this task is related to that

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**